### ISSUE TITLE
unused_ddic - foreign key in another table

### ISSUE THREAD

**Issue by Zeatynis** [abaplint/abaplint#2390]
Hi, shouldn't this rule check also if other tables are referencing to the currently checked table?
![obraz](https://user-images.githubusercontent.com/9119187/152494915-f29af0ca-fecd-4c9f-9443-1340f1844442.png)
![obraz](https://user-images.githubusercontent.com/9119187/152495082-978697ed-d4a9-44b9-8632-a96a5e75c549.png)

**Comment by larshp** [comment]
yes, can you help creating a public example? just a few dummy objects should be enough to reproduce this error?

**Comment by Zeatynis** [comment]
Yes, ofc.
Two Z tables should be enough, right?

**Comment by larshp** [comment]
yea, try :)

**Comment by Zeatynis** [comment]
Funny thing: on version 2.74.61 it's ok 😄 
Here are samples you asked about:
`@EndUserText.label : 'AbapLint test table 1'
@AbapCatalog.enhancement.category : #NOT_EXTENSIBLE
@AbapCatalog.tableCategory : #TRANSPARENT
@AbapCatalog.deliveryClass : #A
@AbapCatalog.dataMaintenance : #RESTRICTED
define table zlint_test1 {
  key client : abap.clnt not null;
  key bukrs  : bukrs not null;
}`
`@EndUserText.label : 'AbapLint test table 2'
@AbapCatalog.enhancement.category : #NOT_EXTENSIBLE
@AbapCatalog.tableCategory : #TRANSPARENT
@AbapCatalog.deliveryClass : #A
@AbapCatalog.dataMaintenance : #RESTRICTED
define table zlint_test2 {
  key client : abap.clnt not null;
  @AbapCatalog.foreignKey.screenCheck : false
  key bukrs  : bukrs not null
    with foreign key zlint_test1
      where client = zlint_test2.client
        and bukrs = zlint_test2.bukrs;

}`

`CLASS zlint_test DEFINITION
  PUBLIC
  FINAL
  CREATE PUBLIC .

  PUBLIC SECTION.
    TYPES ty_bukrs_config TYPE STANDARD TABLE OF zlint_test1 WITH KEY bukrs.
    METHODS get_bukrs_config
      RETURNING VALUE(bukrs_config) TYPE ty_bukrs_config.
  PROTECTED SECTION.
  PRIVATE SECTION.
ENDCLASS.



CLASS zlint_test IMPLEMENTATION.
  METHOD get_bukrs_config.
    SELECT bukrs INTO TABLE bukrs_config
      FROM zlint_test1
      ORDER BY PRIMARY KEY.
    ASSERT sy-subrc <> 0.

  ENDMETHOD.
ENDCLASS.`

[$ZABAPLINT_TEST_CASES_20220204_094524.zip](https://github.com/abaplint/abaplint/files/8001343/ZABAPLINT_TEST_CASES_20220204_094524.zip)

**Comment by larshp** [comment]
how about latest, 2.85.9? does it give the false positive?

**Comment by Zeatynis** [comment]
Docker downloads 2.84.9 as latest and there is a false positive there.

**Comment by larshp** [comment]
argh, the docker build keeps stopping every 60 days, anyhow, triggered again, latest should be up in a bit

**Comment by Zeatynis** [comment]
Yup, on 2.85.9 still false positive

**Comment by larshp** [comment]
anyhow, need the files that abapGit creates, and as of now, TABL are serialized to XML by abapGit

**Comment by Zeatynis** [comment]
What do you mean?

**Comment by Zeatynis** [comment]
Here you can see the effect of the check. There are XMLs that you asked about.
https://github.com/Zeatynis/abaplint-issue/commit/0dc66e613b05c332773dcf5470fc221b54814db0

**Comment by larshp** [comment]
@Zeatynis try again, 2.91.0

**Comment by Zeatynis** [comment]
Same. You can check in my test repo:
https://github.com/Zeatynis/abaplint-issue/actions/runs/2363288530

**Comment by larshp** [comment]
the finding is correct, https://github.com/Zeatynis/abaplint-issue/search?q=zlint_test2

`zlint_test2` is not used anywhere in the other objects?

**Comment by Zeatynis** [comment]
I just rechecked. It's resolved. I must have check it wrong.
